examples/miscellaneous/parallel_backend_benchmarks.py

```diff
@@ -0,0 +1,197 @@
+runtime depends on many factors. It is usually better to experiment rather than
+assuming that it is always a good thing. In some cases it can be highly
+detrimental to performance to run multiple copies of some estimators or
+functions in parallel (see :ref:`oversubscription`).
+
+To run this notebook locally, you may need to install dask and ray. These
+packages can be installed with `pip install dask-ml "ray[default]"`.
+
+For more information see the :ref:`User Guide <parallelism>`.
+
+"""
+
+# %%
+import sys
+import joblib
+
+try:
+    import dask
+except ImportError:
+    print("The package 'dask' is required to run this example.")
+    sys.exit()
+
+try:
+    import ray
+except ImportError:
+    print("The package 'ray' is required to run this example.")
+    sys.exit()
+
+print(joblib.__version__)
+print(dask.__version__)
+print(ray.__version__)
+
+# %%
+# We build a classification task using
+# :class:`~sklearn.datasets.make_classification` and cross-validate an
+# :class:`~sklearn.ensemble.HistGradientBoostingClassifier` with default
+# parameters on top of it.
+
+# %%
+from sklearn.datasets import make_classification
+from sklearn.ensemble import HistGradientBoostingClassifier
+
+X, y = make_classification(n_samples=1_000, random_state=0)
+hist_gbdt = HistGradientBoostingClassifier(random_state=0)
+
+# %%
+# A common setting is to estimate the performance of the model via
+# cross-validation. It is sometimes interesting to set a high number of splits
+# `n_splits` to improve a model's analysis. As a consequence, more computional
+# resourses are needed.
+
+# %%
+from sklearn.model_selection import ShuffleSplit
+
+cv = ShuffleSplit(n_splits=100, random_state=0)
+
+# %%
+# The computional time can still be reduced by optimizing the number of CPUs
+# used via the parameter `n_jobs`.
+
+# In the case of the `~sklearn.model_selection.cross_validate` function,
+# `n_jobs=None` sets the number of workers in a :obj:`joblib.parallel_backend`
+# context. For such function, the parallelization consists in training the
+# estimator and computing the score in parallel over the cross-validation
+# splits.
+
+# %%
+from time import time
+from joblib import parallel_backend
+from sklearn.model_selection import cross_validate
+
+
+def bench(jobs, backend):
+    durations = []
+    msg = f"Benchmarking on {backend}:"
+    print(f"\n{msg}\n" + str("-" * len(msg)))
+    for n_jobs in jobs:
+        with parallel_backend(backend, n_jobs=n_jobs):
+            tic = time()
+            cross_validate(hist_gbdt, X, y, cv=cv)
+            toc = time()
+        durations.append(toc - tic)
+        print(f"n_jobs: {n_jobs:<3} elapsed time: {toc - tic:.3f} sec")
+    return durations
+
+
+# %%
+from ray.util.joblib import register_ray
+from dask.distributed import Client
+
+client = Client(processes=False)  # init local dask client
+ray.shutdown()  # in case there is a previously open ray session
+ray.init(num_cpus=32)
```

Comment:
I think it's good to state the specifications of the machine used to run the benchmarks in the beginning.

Reply:
The results in this example is for 32 cores, so I'll write 32 cores. For others to run it, I suggest a `N_CORES` parameter so it's easy to change.

Reply:
@ArturoAmorQ what does `joblib.cpu_count(only_physical_cores=True)` returns on the machine you used to run this benchmark? 16 or 32?

Based on the timings you report, I have the impression that you machine has even less than 16 physical cores. I find it misleading to report timings for more number of workers larger than the number of cores.

I agree with @thomasjpfan that we should report the number of cores on the machine in a comment in the script and also define the variable to make it easy to change.

Alternatively, you could do what I suggest below and take the max to initialize the local Ray cluster.